src/python/turicreate/toolkits/style_transfer/_sframe_loader.py

```diff
@@ -125,9 +125,13 @@ def provide_data(self):
         return self._provide_data
 
     def _next(self):
-
+        
         if self.cur_epoch == self.num_epochs or len(self.sframe) == 0:
-            raise StopIteration
+            import sys
+            if(sys.version_info.major == 3 and sys.version_info.minor == 7):
```

Comment:
What is the significance of 3 and 7? Could we add a comment explaining the magic numbers or make those into variables?

Reply:
It seems `StopIteration` is still included in Python 3.7. It just get automatically converted to a RuntimeError.
From https://docs.python.org/3.7/library/exceptions.html -
```
Changed in version 3.7: Enable PEP 479 for all code by default: a StopIteration error raised in a generator is transformed into a RuntimeError.
```
So I don't think this change is necessary. @dhivyaaxim - are we missing something?

Reply:
@TobyRoseman  Thanks for your suggestion!

Reply:
@dhivyaaxim - The new comment is not accurate. Python 3.7 does support raising a `StopIteration`. It just converts it to a `RuntimeError`.